### PR TITLE
samples create supports analytics sa

### LINKED ISSUE #67: Update configuration templates to support analytics service account

### PR DESCRIPTION
**Pull request by rockspore**
fix #67

### CODE REVIEW

File: cmd/samples/samples.go [hunk 1]
```diff
@@ -137,8 +138,15 @@ func (s *samples) loadConfig() error {
 	s.Env = s.ServerConfig.Tenant.EnvName
 	s.Namespace = s.ServerConfig.Global.Namespace
 
+	// handle configs for analytics-related credential
 	if s.ServerConfig.IsGCPManaged() {
-		s.EncodedName = provision.EnvScopeEncodedName(s.Org, s.Env)
+		s.IsGCPManaged = true
+		// SA credentials supersede the fluentd enpoint
```
Comment (theganyo) [inline cmd/samples/samples.go:144]: Should we log a warning if both exist and the fluentd endpoint is being superceded?

File: cmd/samples/samples_test.go [hunk 2]
```diff
@@ -31,9 +31,10 @@ import (
 func TestCreateNativeConfigs(t *testing.T) {
 	print := testutil.Printer("TestCreateNativeConfigs")
 
-	defer os.RemoveAll("./native")
+	tmpDir := "./native"
```
Comment (theganyo) [inline cmd/samples/samples_test.go:34]: You have a number of these hard-coded tmpDirs... should we be using `ioutil.TempDir`?

File: cmd/samples/samples.go (hunk 1 above)
Comment (rockspore) [reply cmd/samples/samples.go:144]: Sounds good.

File: cmd/samples/samples_test.go (hunk 2 above)
Comment (rockspore) [reply cmd/samples/samples_test.go:34]: The reason is TempDir creates the directory while we would like the CLI to create the directory. Does a `ioutil.TempDir()` followed by a `os.RemoveAll()` (in addition to the deferred one) sound better to you?

Comment (theganyo) [reply cmd/samples/samples_test.go]: Yes, in general I think any directory created during testing should be created in a tempDir.

Comment (rockspore) [reply cmd/samples/samples_test.go]: Makes sense. It has been addressed in the latest commit.

**Review by theganyo — APPROVED** [review]

**Pull request merged**